File: src/main/java/com/freya02/botcommands/api/application/slash/annotations/ChannelTypes.java
package com.freya02.botcommands.api.application.slash.annotations;


import com.freya02.botcommands.api.application.annotations.AppOption;
import net.dv8tion.jda.api.entities.channel.ChannelType;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

/**
 * Allows you to set the desired channel types for this {@link AppOption app option}
 */
@Retention(RetentionPolicy.RUNTIME)
@Target({ElementType.PARAMETER})
public @interface ChannelTypes {
	ChannelType[] value();
}

File: BotCommands-commands/app/src/main/kotlin/io/github/freya022/botcommands/api/commands/application/slash/annotations/ChannelTypes.kt
package io.github.freya022.botcommands.api.commands.application.slash.annotations

import net.dv8tion.jda.api.entities.channel.ChannelType
import net.dv8tion.jda.api.entities.channel.concrete.TextChannel

/**
 * Sets the desired channel types for this parameter.
 *
 * This works for annotated commands as well as code-declared commands
 *
 * You can alternatively use a specific channel type,
 * such as [TextChannel] to automatically restrict the channel type.
 *
 * ### Merging
 * This annotation can be merged if found with other meta-annotations.
 * Keep in mind that a *direct* annotation overrides all meta-annotations.
 */
@Target(AnnotationTarget.VALUE_PARAMETER)
@Retention(AnnotationRetention.RUNTIME)
annotation class ChannelTypes(vararg val value: ChannelType)
